FizzBuzz/src/test/java/rocks/kata/FizzBuzzTest.java

```diff
@@ -1,49 +1,145 @@
+
+        private static List<FizzBuzzTestCase> createTestCases() {
+            List<FizzBuzz> listOfFizzBuzzObjects = Arrays.asList(
+                    new FizzBuzzDivisibilityRules(),
+                    new FizzBuzzLiterals(),
+                    new FizzBuzzModulo(),
+                    new FizzBuzzPattern());
+            List<Integer> listOfNumbersDivisibleBy3 = Arrays.asList(3, 12, 3 * 100003);
+            List<Integer> listOfNumbersDivisibleBy5 = Arrays.asList(5, 25, 5 * 100003);
+            List<Integer> listOfNumbersDivisibleBy15 = Arrays.asList(15, 15 * 15, 15 * 100003);
+            List<Integer> listOfNumbersNotDivisibleBy3or5 = Arrays.asList(2, 19, 100003);
+
+            List<FizzBuzzTestCase> testCases = new ArrayList<>();
+
+            for (FizzBuzz fizzBuzzObject : listOfFizzBuzzObjects) {
+                for (int number : listOfNumbersDivisibleBy3) {
+                    testCases.add(createFizzBuzzTestCase(number, fizzBuzzObject, "Fizz"));
+                }
+                for (int number : listOfNumbersDivisibleBy5) {
+                    testCases.add(createFizzBuzzTestCase(number, fizzBuzzObject, "Buzz"));
+                }
+                for (int number : listOfNumbersDivisibleBy15) {
+                    testCases.add(createFizzBuzzTestCase(number, fizzBuzzObject, "FizzBuzz"));
+                }
+                for (int number : listOfNumbersNotDivisibleBy3or5) {
+                    testCases.add(createFizzBuzzTestCase(number, fizzBuzzObject, Integer.toString(number)));
+                }
+
+            }
+            return testCases;
+        }
+
+        private static FizzBuzzTestCase createFizzBuzzTestCase(int number, FizzBuzz fizzBuzzInstance,
+                String expectedResult) {
+            return new FizzBuzzTestCase(generateTestDescription(fizzBuzzInstance, number), number, fizzBuzzInstance,
+                    expectedResult);
+        }
+
+        private static String generateTestDescription(FizzBuzz fizzBuzzObject, int number) {
+            return "Object " + fizzBuzzObject.getClass().getSimpleName() + " and number " + number;
+        }
     }
 
-    @ParameterizedTest
-    @ValueSource(ints = { 2, 19, 98 * 100003 })
-    void shouldReturnNumberIfNotDivisibleByThreeOrFive(int number) {
-        assertTrue(fizzPattern.returnFizzBuzzOrNumber(number).equals(Integer.toString(number)));
-        assertTrue(fizzStringBuilder.returnFizzBuzzOrNumber(number).equals(Integer.toString(number)));
-        assertTrue(fizzLiterals.returnFizzBuzzOrNumber(number).equals(Integer.toString(number)));
+    static private class FizzBuzzTestCaseParameterResolver implements ParameterResolver {
+
+        private FizzBuzzTestCase testCase;
+
+        private FizzBuzzTestCaseParameterResolver(FizzBuzzTestCase testCase) {
+            this.testCase = testCase;
+        }
+
+        @Override
+        public Object resolveParameter(ParameterContext arg0, ExtensionContext arg1)
```

Comment:
automatycznie generowane nazwy argumentów zastąp ładniejszymi, np. `paramContext` i `extensionContext` - teraz nie są używane, ale może będą w przyszłości (poniżej to samo)